Change stddev to link to stddev_samp instead of stddev_pop

matriv opened the pull request:
Follows: #17543
Follows: #17910
Relates: #15638

- Additionally, improve data type tests for the functions.

seut commented on docs/appendices/release-notes/6.0.0.rst:
```diff
@@ -219,6 +219,11 @@ Breaking Changes
+ - :ref:`aggregation-stddev` changed and is now an alias for
+   :ref:`aggregation-stddev-samp`, to match PostgreSQL behavior. In order to
+   calculate the population standard deviation, :ref:`aggregation-stddev-pop`
+   should be used.
```
```suggestion
   must be used.
```

seut reviewed (approved):
👍 Maybe adding a simple test which verifies that `stddev` is an alias to `stddev_samp` would be valuable.

matriv commented:
> 👍 Maybe adding a simple test which verifies that `stddev` is an alias to `stddev_samp` would be valuable.

thx, added both in unit test, and reverted one of the integ tests so the expected result is changed since `stddev` now uses `stddev_samp`.

matriv commented:
@mergifyio requeue
